File: 251002/Korti/publisher/src/entities/Article.ts
import {
  Column,
  Entity,
  JoinColumn,
  JoinTable,
  ManyToMany,
  ManyToOne,
  OneToMany,
  PrimaryGeneratedColumn,
} from 'typeorm';
import { Editor } from './Editor';
import { Sticker } from './Sticker';
import { Note } from './Note';
@Entity({ name: 'tbl_article' })
export class Article {
  @PrimaryGeneratedColumn()
  id: number;

  @Column({ name: 'editor_Id', nullable: false })
  editorId: number;

  @Column({ nullable: false })
  title: string;

  @Column({ nullable: false })
  content: string;

  @Column({ type: 'timestamp', default: () => 'CURRENT_TIMESTAMP' })
  created: Date;

  @Column({ type: 'timestamp', nullable: true })
  modified: Date | null;

  @OneToMany(() => Note, (note) => note.article, { onDelete: 'CASCADE' })
  notes: Note[];

  @ManyToOne(() => Editor, { onDelete: 'CASCADE' })
  @JoinColumn({ name: 'editor_Id' })
  editor: Editor;

  @ManyToMany(() => Sticker, { onDelete: 'CASCADE' })
  @JoinTable({
    name: 'tbl_sticker_article',
    joinColumn: { name: 'articleId' },
    inverseJoinColumn: { name: 'stickerId' },
  })
  stickers: Sticker[];
}

File: 251003/Stasevich/Publisher/Entity/Article.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace WebApplication1.Entity
{
    public class Article
    {
        public long Id { get; set; }

        [Column("user_id")]
        public long UserId { get; set; }
        public string Title { get; set; } = string.Empty;
        public string Content { get; set; } = string.Empty;
        public DateTime Created { get; set; } = DateTime.UtcNow;
        public DateTime Modified { get; set; } = DateTime.UtcNow;

        public User User { get; set; } = null!;
        public List<Note> Notes { get; set; } = new();
        public List<Sticker> Stickers { get; set; } = new();
    }
}
